bin/civihr-install

```diff
@@ -0,0 +1,820 @@
+    civihr_assertvars default_cache_setup PRJDIR
+    git_cache_setup "https://github.com/civicrm/civihr.git" "$CACHE_DIR/civicrm/civihr.git"
+    git_cache_setup "https://github.com/civicrm/civicrm-core.git" "$CACHE_DIR/civicrm/civicrm-core.git"
+    git_cache_setup "https://github.com/civicrm/civicrm-drupal.git"  "$CACHE_DIR/civicrm/civicrm-drupal.git"
+    git_cache_setup "https://github.com/civicrm/civicrm-packages.git" "$CACHE_DIR/civicrm/civicrm-packages.git"
+    git_cache_setup "https://github.com/eileenmcnaughton/civicrm_developer.git" "$CACHE_DIR/eileenmcnaughton/civicrm_developer.git"
+  fi
+}
+
+###############################################################################
+## Parse options
+function civihr_parse() {
+
+  civihr_mkdir "$TMPDIR" "$BLDDIR"
+
+  # Checking for site-name parameter.
+  [ "${1::1}" == "-" ]  && echo "[[Please provide site-name]]" && civihr_app_usage && exit 1
+  
+  [ -z "$SITE_NAME" ]  && SITE_NAME="$1"
+
+  while [ -n "$1" ] ; do
+    OPTION="$1"
+    shift
+    case "$OPTION" in
+      -h|--help|-?)
+        civihr_app_usage
+        ;;
+
+      --dbuser)
+        [ "${1::1}" == "-" ] && echo "[[Missing value for option --dbuser.]]" && civihr_app_usage && exit 1
+        DB_USER="$1"
+        shift
+        ;;
+      --dbpass)
+        [ "${1::1}" == "-" ] && echo "[[Missing value for option --dbpass.]]" && civihr_app_usage && exit 1
+        DB_PASS="$1"
+        shift
+        ;;
+      --dbport)
+        DB_PORT="$1"
+        shift
+        ;;
+
+      --cmsdb)
+        [ "${1::1}" == "-" ] && echo "[[Missing value for option --cmsdb.]]" && civihr_app_usage && exit 1
+        CMSdbName="$1"
+        shift
+        ;;
+
+      --crmdb)
+        [ "${1::1}" == "-" ] && echo "[[Missing value for option --crmdb.]]" && civihr_app_usage && exit 1
+        CIVIdbName="$1"
+        shift
+        ;;
+
+      --url)
+        CMS_URL="$1"
+        shift
+        ;;
+
+      *)
+         if [ "${OPTION::1}" == "-" ]; then
+           echo "[[Unrecognized option: $OPTION]]"
+           civihr_app_usage
+           exit
+         fi
+        ;;
+    esac
+  done
+
+# Checking for all required options/arguments.
+[ -z "$DB_USER" ] || [ -z "$DB_PASS" ] || [ -z "$CMSdbName" ] || [ -z "$CIVIdbName" ] || [ -z "$CMS_URL" ] && echo "[[Please provide missing options/agruments.]]" && civihr_app_usage && exit 99
+
+# Assigning values.
+[ -z "$SITE_NAME" ]          && civihr_app_usage && exit 99
+[ -z "$WEB_ROOT" ]           && WEB_ROOT="$BLDDIR/$SITE_NAME"
+[ -z "$CMS_ROOT" ]           && CMS_ROOT="$WEB_ROOT"
+[ -z "$CIVI_SITE_KEY" ]      && CIVI_SITE_KEY=$(civihr_makepasswd 16)
+[ -z "$ADMIN_PASS" ]         && ADMIN_PASS=$(civihr_makepasswd 12)
+[ -z "$CMS_TITLE" ]          && CMS_TITLE="$SITE_NAME"
+[ -z "$CMS_HOSTNAME" ]       && CMS_HOSTNAME=$(php -r '$p = parse_url($argv[1]); echo $p["host"];' "$CMS_URL")
+[ -z "$CMS_PORT" ]           && CMS_PORT=$(php -r '$p = parse_url($argv[1]); echo $p["port"];' "$CMS_URL")
```

Comment:
This line is issuing a PHP notice when port is not defined:
> PHP Notice:  Undefined index: port in Command line code on line 1

We could have a default value if it's not set, or try something like:
`[ -z "$CMS_PORT" ]           && CMS_PORT=$(php -r '$p = parse_url($argv[1]); if (isset($p["port"])) {echo $p["port"];}' "$CMS_URL")`